### PR TITLE
automate nudging of dependency alerts

### LINKED ISSUE #783: automate nudging of security upgrades for dependencies
## Background information

As [part of OpsRotation](https://github.com/18F/tts-tech-portfolio/blob/master/.github/ISSUE_TEMPLATE/ops.md#github), I was jumping into various Slack channels asking teams to merge pull requests from Dependabot/Snyk, and thought to myself "this is really a pain." When we had considered how to do things like this in the past, an obstacle had been figuring out how to map repositories to Slack channels in a consistent way. Trying to do so manually would be a major obstacle.

My strategy for finding the right channel was searching the repository URL in Slack and then posting in whatever channel it was mentioned the most. That made me realize: Rather than relying on explicit mapping from repositories to channels, we could use the implicit mapping through those Slack searchers. This will always be up-to-date, and accommodates organic movement of teams between channels without metadata.

## User stories

- As someone responsible for security in TTS, I want to minimize effort ensuring that systems have up-to-date application dependencies.

## Acceptance criteria

- We have eliminated or significantly reduced the amount of time required for the OpsRotation person to nudge teams about open security upgrades

---

The assignee should add some checkboxes as a "sketch" of the steps to complete, which may evolve.

### PR DESCRIPTION
**Pull request by afeld**
Closes #783. Built off of https://github.com/18F/tts-tech-portfolio/pull/824.

## TODOs

- [x] Add usage documentation
- [x] Update the Ops Rotation Checklist
- [x] Look into why [Tock got notified about alerts](https://gsa-tts.slack.com/archives/C1JFYCX3P/p1600696476000300) even though they didn't have any open ones

### CODE REVIEW

File: slack-inspector/repos.graphql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+query repos($org: String!, $cursor: String) {
+  organization(login: $org) {
+    repositories(first: 100, after: $cursor) {
+      edges {
+        node {
+          nameWithOwner
+          isArchived
+          vulnerabilityAlerts {
+            totalCount
```
Comment (Jkrzy) [inline slack-inspector/repos.graphql:9]: I _think_ this returns the count of **all** `vulnerabilityAlerts` regardless of open/closed/dismissed status.

Comment (afeld) [reply slack-inspector/repos.graphql]: Good catch, thanks!

**Comment by afeld** [comment]
Ok, this is ready! Some backstory:

For https://github.com/18F/tts-tech-portfolio/issues/122, I wrote some code to retrieve all our `-partner` channels (https://github.com/18F/tts-tech-portfolio/pull/824). That was a one-time need, but in doing so I realized that the code would be useful for finding channels corresponding to GitHub repositories. I was on OpsRotation at the time, so did a spike on sending notifications for open security alerts. It worked better than expected. Continued to refine, and here we are.

The hope is that it will take the time OpsRotation spends on those nudges (~30 minutes every other week) and take that to a couple minutes, then zero by fully automating it. More importantly, it gives us something to build off of for other automation around GitHub repositories / Slack.

This should probably merge with [Charlie](https://github.com/18F/18f-bot) at some point. That only occurred to me after starting in this repository, and there are other structural reasons it would take some work.

**Review by Jkrzy — APPROVED** [review]
Looks good to me and is a great foundation for future work, thanks @afeld 
- [x] Documentation, script and rotation, updates
- [x] Great test coverage, executing in CI, and passing
- [x] A bit of manual testing

**Review by JJediny — APPROVED** [review]
wow nice work lgtm

**Pull request merged**